site/docs/nessie.md

```diff
@@ -0,0 +1,120 @@
+* `uri`. This is the Nessie server base uri. Eg `http://localhost:19120/api/v1`.
+* `ref` (optional). This is the Nessie branch or tag you want to work in.
+
+To run directly in Java this looks like:
+
+``` java
+Map<String, String> options = new HashMap<>();
+options.put("warehouse", "/path/to/warehouse");
+options.put("ref", "main");
+options.put("uri", "https://localhost:19120/api/v1");
+Catalog nessieCatalog = CatalogUtil.loadCatalog("org.apache.iceberg.nessie.NessieCatalog", "nessie", hadoopConfig, options);
+```
+
+and in Spark:
+
+``` java
+conf.set("spark.sql.catalog.nessie.warehouse", "/path/to/warehouse");
+conf.set("spark.sql.catalog.nessie.uri", "http://localhost:19120/api/v1")
+conf.set("spark.sql.catalog.nessie.ref", "main")
+conf.set("spark.sql.catalog.nessie.catalog-impl", "org.apache.iceberg.nessie.NessieCatalog")
+conf.set("spark.sql.catalog.nessie", "org.apache.iceberg.spark.SparkCatalog")
+```
+
+Once you have a Nessie catalog you have access to your entire Nessie repo. You can then perform create/delete/merge
+operations on branches and perform commits on branches. Each iceberg table in a Nessie Catalog is identified by an
+arbitrary length namespace and table name (eg `data.base.name.table`). These namespaces are implicit and don't need to
+be created separately. Any transaction on a Nessie enabled Iceberg table is a single commit in Nessie. Nessie commits
+can encompass an arbitrary number of actions on an arbitrary number of tables, however in Iceberg this will be limited
+to the set of single table transactions currently available.
+
+Further operations such as merges, viewing the commit log or diffs are performed by direct interaction with the
+`NessieClient` in java or by using the python client or cli. See [Nessie CLI](https://projectnessie.org/tools/cli/) for
+more details on the CLI and [Spark Guide](https://projectnessie.org/tools/spark/) for a more complete description of 
+Nessie functionality.
+
+## Nessie and Iceberg
+
+For most cases Nessie acts just like any other Catalog for Iceberg: providing a logical organization of a set of tables
+and providing atomicity to transactions. However using Nessie opens up other interesting possibilities.
+
+### Loosely coupled multi-table transactions
+
+By creating a branch and performing a set of operations on that branch you can approximate a multi-table transaction.
+A sequence of commits can be performed on the newly created branch and then merged back into the main branch atomically.
+This gives the appearance of a series of connected changes being exposed to the main branch atomically. Downstream
+consumers will see this as a multi-table transaction on the database. Unlike a traditional database transaction this
+branch based transaction doesn't have a single owner. For example, a series of globally distributed Spark applications
+can simultaneously update the branch and the merge back to the main branch will expose all these changes atomically.
```

Comment:
I think this section should be a bit more clear about what is supported by Nessie. Claiming support for multi-table transactions seems misleading to me, because it's really the equivalent of a "fast-forward" operation and not what I would think of normally as a transaction. It would help to have a section that makes that clear in terms people familiar with `git` would understand.

Reply:
cool, thanks for the feedback. I have softened the language. I am working on a 'nessie for devs' post on the nessie docs site and will add a link to it in these docs when up. I felt that keeping the nessie specific stuff relatively light here and providing a lot of links would be better than adding tons of Nessie stuff to iceberg's docs.